src/capi.rs

```diff
@@ -0,0 +1,212 @@
+use libc::c_void;
+use backend;
+use cubeb;
+
+extern "C" fn capi_init(c: *mut *mut cubeb::Context, context_name: *const i8) -> i32
+{
+    unsafe {
```

Comment:
I think that's a fair assessment. I've added a lot of unsafe blocks around the PA calls in the backend, but I want to leave those for now, as I've explained, I intend to replace the raw FFI calls with more ergonomic rust wrappers.